File: src/main/java/io/github/svegon/capi/mixininterface/IRenderTickCounter.java
package io.github.svegon.capi.mixininterface;

public interface IRenderTickCounter {
    long getPrevTimeMillis();

    void setPrevTimeMillis(long prevTimeMillis);

    float getTickTime();

    void setTickTime(float tickTime);

    default float getTpS() {
        return 1000 / getTickTime();
    }

    default void setTpS(float tps) {
        setTickTime(1000 / tps);
    }
}

File: src/main/kotlin/io/github/svegon/mclientapi/mixininterface/IRenderTickCounter.kt
package io.github.svegon.capi.mixininterface

interface IRenderTickCounter {
    var prevTimeMillis: Long

    var tickTime: Float

    var tpS: Float
        get() = 1000 / tickTime
        set(tps) {
            tickTime = 1000 / tps
        }
}
